src/types/onyx/Policy.ts

```diff
@@ -68,18 +69,101 @@ type TaxRatesWithDefault = {
     taxes: TaxRates;
 };
 
-// These types are for the Integration connections for a policy (eg. Quickbooks, Xero, etc).
-// This data is not yet used in the codebase which is why it is given a very generic type, but the data is being put into Onyx for future use.
-// Once the data is being used, these types should be defined appropriately.
-type ConnectionLastSync = Record<string, unknown>;
-type ConnectionData = Record<string, unknown>;
-type ConnectionConfig = Record<string, unknown>;
+type ConnectionLastSync = {
+    successfulDate?: string;
+    errorDate?: string;
+    isSuccessful: boolean;
+    source: 'DIRECT' | 'EXPENSIFYWEB' | 'EXPENSIFYAPI' | 'AUTOSYNC' | 'AUTOAPPROVE';
+};
+
+type Account = {
+    glCode?: string;
+    name: string;
+    currency: string;
+    id: number;
+};
+
+type Employee = {
+    id: number;
+    firstName: string;
+    lastName: string;
+    name: string;
+    email: string;
+};
+
+type Vendor = {
+    id: number;
+    name: string;
+    currency: string;
+    email: string;
+};
+
+type ConnectionData = {
+    country: string;
+    edition: string;
+    homeCurrency: string;
+    isMultiCurrencyEnabled: boolean;
+
+    journalEntryAccounts: Account[];
+    bankAccounts: Account[];
+    creditCards: Account[];
+    accountsReceivable: Account[];
+    accountsPayable: Account[];
+    otherCurrentAssetAccounts: Account[];
+
+    taxCodes: string[];
+    employees: Employee[];
+    vendors: Vendor[];
+};
+
+type IntegrationEntity = 'NONE' | 'DEFAULT' | 'TAG' | 'REPORT_FIELD';
+
+type ConnectionConfig = {
+    realmId: number;
+    credentials: {
```

Comment:
Do we need to add the definition for this in the client? It won't need it.

Reply:
we will send back the data as it's stored in the backend, and I think as long as we have data, we should define type for it 🤔

Reply:
I removed `credentials`